pkgs/development/octave-modules/tisean/default.nix

```diff
@@ -0,0 +1,33 @@
+{ buildOctavePackage
+, stdenv
+, fetchurl
+# Octave dependencies
+, signal # >= 1.3.0
+# Build dependencies
+, gfortran
+}:
+
+buildOctavePackage rec {
+  pname = "tisean";
+  version = "0.2.3";
+
+  src = fetchurl {
+    url = "https://octave.sourceforge.io/download.php?package=${pname}-${version}.tar.gz";
+    sha256 = "0nc2d9h91glxzmpizxdrc2dablw4bqhqhzs37a394c36myk4xjdv";
+  };
+
+  buildInputs = [
+    gfortran
```

Comment:
If this is used as a compiler shouldn't it go to nativeBuildInputs?

Reply:
It probably should, yes. I when writing most of these files, I was unsure of whether to use `buildInputs` or `nativeBuildInputs`. I understand that the difference comes to the issue of cross-compilation (`nativeBuildInputs` are tools that are build using the architecture of the system performing the building, which may not be the same as the system running the package/program), but am still unsure about when to make a package a `buildInput` or a `nativeBuildInput`.